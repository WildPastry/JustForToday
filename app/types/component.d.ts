interface IExternalLink {
  children: JSX.Element;
  href: string;
  style: {};
}

interface IEditScreenInfo {
  path: string;
}
