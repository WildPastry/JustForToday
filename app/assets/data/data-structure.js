const dataStructure = 'data-structure';
return dataStructure;
