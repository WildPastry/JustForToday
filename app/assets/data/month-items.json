[
  {
    "id": "Jan",
    "name": "January",
    "days": [
      { "id": "0101", "name": "1" },
      { "id": "0201", "name": "2" },
      { "id": "0301", "name": "3" },
      { "id": "0401", "name": "4" },
      { "id": "0501", "name": "5" },
      { "id": "0601", "name": "6" },
      { "id": "0701", "name": "7" },
      { "id": "0801", "name": "8" },
      { "id": "0901", "name": "9" },
      { "id": "1001", "name": "10" },
      { "id": "1101", "name": "11" },
      { "id": "1201", "name": "12" },
      { "id": "1301", "name": "13" },
      { "id": "1401", "name": "14" },
      { "id": "1501", "name": "15" },
      { "id": "1601", "name": "16" },
      { "id": "1701", "name": "17" },
      { "id": "1801", "name": "18" },
      { "id": "1901", "name": "19" },
      { "id": "2001", "name": "20" },
      { "id": "2101", "name": "21" },
      { "id": "2201", "name": "22" },
      { "id": "2301", "name": "23" },
      { "id": "2401", "name": "24" },
      { "id": "2501", "name": "25" },
      { "id": "2601", "name": "26" },
      { "id": "2701", "name": "27" },
      { "id": "2801", "name": "28" },
      { "id": "2901", "name": "29" },
      { "id": "3001", "name": "30" },
      { "id": "3101", "name": "31" }
    ]
  },
  {
    "id": "Feb",
    "name": "Feburary",
    "days": [
      { "id": "0102", "name": "1" },
      { "id": "0202", "name": "2" },
      { "id": "0302", "name": "3" },
      { "id": "0402", "name": "4" },
      { "id": "0502", "name": "5" },
      { "id": "0602", "name": "6" },
      { "id": "0702", "name": "7" },
      { "id": "0802", "name": "8" },
      { "id": "0902", "name": "9" },
      { "id": "1002", "name": "10" },
      { "id": "1102", "name": "11" },
      { "id": "1202", "name": "12" },
      { "id": "1302", "name": "13" },
      { "id": "1402", "name": "14" },
      { "id": "1502", "name": "15" },
      { "id": "1602", "name": "16" },
      { "id": "1702", "name": "17" },
      { "id": "1802", "name": "18" },
      { "id": "1902", "name": "19" },
      { "id": "2002", "name": "20" },
      { "id": "2102", "name": "21" },
      { "id": "2202", "name": "22" },
      { "id": "2302", "name": "23" },
      { "id": "2402", "name": "24" },
      { "id": "2502", "name": "25" },
      { "id": "2602", "name": "26" },
      { "id": "2702", "name": "27" },
      { "id": "2802", "name": "28" },
      { "id": "2902", "name": "29" }
    ]
  },
  {
    "id": "Mar",
    "name": "March",
    "days": [
      { "id": "0103", "name": "1" },
      { "id": "0203", "name": "2" },
      { "id": "0303", "name": "3" },
      { "id": "0403", "name": "4" },
      { "id": "0503", "name": "5" },
      { "id": "0603", "name": "6" },
      { "id": "0703", "name": "7" },
      { "id": "0803", "name": "8" },
      { "id": "0903", "name": "9" },
      { "id": "1003", "name": "10" },
      { "id": "1103", "name": "11" },
      { "id": "1203", "name": "12" },
      { "id": "1303", "name": "13" },
      { "id": "1403", "name": "14" },
      { "id": "1503", "name": "15" },
      { "id": "1603", "name": "16" },
      { "id": "1703", "name": "17" },
      { "id": "1803", "name": "18" },
      { "id": "1903", "name": "19" },
      { "id": "2003", "name": "20" },
      { "id": "2103", "name": "21" },
      { "id": "2203", "name": "22" },
      { "id": "2303", "name": "23" },
      { "id": "2403", "name": "24" },
      { "id": "2503", "name": "25" },
      { "id": "2603", "name": "26" },
      { "id": "2703", "name": "27" },
      { "id": "2803", "name": "28" },
      { "id": "2903", "name": "29" },
      { "id": "3003", "name": "30" },
      { "id": "3103", "name": "31" }
    ]
  },
  {
    "id": "Apr",
    "name": "April",
    "days": [
      { "id": "0104", "name": "1" },
      { "id": "0204", "name": "2" },
      { "id": "0304", "name": "3" },
      { "id": "0404", "name": "4" },
      { "id": "0504", "name": "5" },
      { "id": "0604", "name": "6" },
      { "id": "0704", "name": "7" },
      { "id": "0804", "name": "8" },
      { "id": "0904", "name": "9" },
      { "id": "1004", "name": "10" },
      { "id": "1104", "name": "11" },
      { "id": "1204", "name": "12" },
      { "id": "1304", "name": "13" },
      { "id": "1404", "name": "14" },
      { "id": "1504", "name": "15" },
      { "id": "1604", "name": "16" },
      { "id": "1704", "name": "17" },
      { "id": "1804", "name": "18" },
      { "id": "1904", "name": "19" },
      { "id": "2004", "name": "20" },
      { "id": "2104", "name": "21" },
      { "id": "2204", "name": "22" },
      { "id": "2304", "name": "23" },
      { "id": "2404", "name": "24" },
      { "id": "2504", "name": "25" },
      { "id": "2604", "name": "26" },
      { "id": "2704", "name": "27" },
      { "id": "2804", "name": "28" },
      { "id": "2904", "name": "29" },
      { "id": "3004", "name": "30" }
    ]
  },
  {
    "id": "May",
    "name": "May",
    "days": [
      { "id": "0105", "name": "1" },
      { "id": "0205", "name": "2" },
      { "id": "0305", "name": "3" },
      { "id": "0405", "name": "4" },
      { "id": "0505", "name": "5" },
      { "id": "0605", "name": "6" },
      { "id": "0705", "name": "7" },
      { "id": "0805", "name": "8" },
      { "id": "0905", "name": "9" },
      { "id": "1005", "name": "10" },
      { "id": "1105", "name": "11" },
      { "id": "1205", "name": "12" },
      { "id": "1305", "name": "13" },
      { "id": "1405", "name": "14" },
      { "id": "1505", "name": "15" },
      { "id": "1605", "name": "16" },
      { "id": "1705", "name": "17" },
      { "id": "1805", "name": "18" },
      { "id": "1905", "name": "19" },
      { "id": "2005", "name": "20" },
      { "id": "2105", "name": "21" },
      { "id": "2205", "name": "22" },
      { "id": "2305", "name": "23" },
      { "id": "2405", "name": "24" },
      { "id": "2505", "name": "25" },
      { "id": "2605", "name": "26" },
      { "id": "2705", "name": "27" },
      { "id": "2805", "name": "28" },
      { "id": "2905", "name": "29" },
      { "id": "3005", "name": "30" },
      { "id": "3105", "name": "31" }
    ]
  },
  {
    "id": "Jun",
    "name": "June",
    "days": [
      { "id": "0106", "name": "1" },
      { "id": "0206", "name": "2" },
      { "id": "0306", "name": "3" },
      { "id": "0406", "name": "4" },
      { "id": "0506", "name": "5" },
      { "id": "0606", "name": "6" },
      { "id": "0706", "name": "7" },
      { "id": "0806", "name": "8" },
      { "id": "0906", "name": "9" },
      { "id": "1006", "name": "10" },
      { "id": "1106", "name": "11" },
      { "id": "1206", "name": "12" },
      { "id": "1306", "name": "13" },
      { "id": "1406", "name": "14" },
      { "id": "1506", "name": "15" },
      { "id": "1606", "name": "16" },
      { "id": "1706", "name": "17" },
      { "id": "1806", "name": "18" },
      { "id": "1906", "name": "19" },
      { "id": "2006", "name": "20" },
      { "id": "2106", "name": "21" },
      { "id": "2206", "name": "22" },
      { "id": "2306", "name": "23" },
      { "id": "2406", "name": "24" },
      { "id": "2506", "name": "25" },
      { "id": "2606", "name": "26" },
      { "id": "2706", "name": "27" },
      { "id": "2806", "name": "28" },
      { "id": "2906", "name": "29" },
      { "id": "3006", "name": "30" },
      { "id": "3106", "name": "31" }
    ]
  },
  {
    "id": "Jul",
    "name": "July",
    "days": [
      { "id": "0107", "name": "1" },
      { "id": "0207", "name": "2" },
      { "id": "0307", "name": "3" },
      { "id": "0407", "name": "4" },
      { "id": "0507", "name": "5" },
      { "id": "0607", "name": "6" },
      { "id": "0707", "name": "7" },
      { "id": "0807", "name": "8" },
      { "id": "0907", "name": "9" },
      { "id": "1007", "name": "10" },
      { "id": "1107", "name": "11" },
      { "id": "1207", "name": "12" },
      { "id": "1307", "name": "13" },
      { "id": "1407", "name": "14" },
      { "id": "1507", "name": "15" },
      { "id": "1607", "name": "16" },
      { "id": "1707", "name": "17" },
      { "id": "1807", "name": "18" },
      { "id": "1907", "name": "19" },
      { "id": "2007", "name": "20" },
      { "id": "2107", "name": "21" },
      { "id": "2207", "name": "22" },
      { "id": "2307", "name": "23" },
      { "id": "2407", "name": "24" },
      { "id": "2507", "name": "25" },
      { "id": "2607", "name": "26" },
      { "id": "2707", "name": "27" },
      { "id": "2807", "name": "28" },
      { "id": "2907", "name": "29" },
      { "id": "3007", "name": "30" },
      { "id": "3107", "name": "31" }
    ]
  },
  {
    "id": "Aug",
    "name": "August",
    "days": [
      { "id": "0108", "name": "1" },
      { "id": "0208", "name": "2" },
      { "id": "0308", "name": "3" },
      { "id": "0408", "name": "4" },
      { "id": "0508", "name": "5" },
      { "id": "0608", "name": "6" },
      { "id": "0708", "name": "7" },
      { "id": "0808", "name": "8" },
      { "id": "0908", "name": "9" },
      { "id": "1008", "name": "10" },
      { "id": "1108", "name": "11" },
      { "id": "1208", "name": "12" },
      { "id": "1308", "name": "13" },
      { "id": "1408", "name": "14" },
      { "id": "1508", "name": "15" },
      { "id": "1608", "name": "16" },
      { "id": "1708", "name": "17" },
      { "id": "1808", "name": "18" },
      { "id": "1908", "name": "19" },
      { "id": "2008", "name": "20" },
      { "id": "2108", "name": "21" },
      { "id": "2208", "name": "22" },
      { "id": "2308", "name": "23" },
      { "id": "2408", "name": "24" },
      { "id": "2508", "name": "25" },
      { "id": "2608", "name": "26" },
      { "id": "2708", "name": "27" },
      { "id": "2808", "name": "28" },
      { "id": "2908", "name": "29" },
      { "id": "3008", "name": "30" },
      { "id": "3108", "name": "31" }
    ]
  },
  {
    "id": "Sep",
    "name": "September",
    "days": [
      { "id": "0109", "name": "1" },
      { "id": "0209", "name": "2" },
      { "id": "0309", "name": "3" },
      { "id": "0409", "name": "4" },
      { "id": "0509", "name": "5" },
      { "id": "0609", "name": "6" },
      { "id": "0709", "name": "7" },
      { "id": "0809", "name": "8" },
      { "id": "0909", "name": "9" },
      { "id": "1009", "name": "10" },
      { "id": "1109", "name": "11" },
      { "id": "1209", "name": "12" },
      { "id": "1309", "name": "13" },
      { "id": "1409", "name": "14" },
      { "id": "1509", "name": "15" },
      { "id": "1609", "name": "16" },
      { "id": "1709", "name": "17" },
      { "id": "1809", "name": "18" },
      { "id": "1909", "name": "19" },
      { "id": "2009", "name": "20" },
      { "id": "2109", "name": "21" },
      { "id": "2209", "name": "22" },
      { "id": "2309", "name": "23" },
      { "id": "2409", "name": "24" },
      { "id": "2509", "name": "25" },
      { "id": "2609", "name": "26" },
      { "id": "2709", "name": "27" },
      { "id": "2809", "name": "28" },
      { "id": "2909", "name": "29" },
      { "id": "3009", "name": "30" }
    ]
  },
  {
    "id": "Oct",
    "name": "October",
    "days": [
      { "id": "0110", "name": "1" },
      { "id": "0210", "name": "2" },
      { "id": "0310", "name": "3" },
      { "id": "0410", "name": "4" },
      { "id": "0510", "name": "5" },
      { "id": "0610", "name": "6" },
      { "id": "0710", "name": "7" },
      { "id": "0810", "name": "8" },
      { "id": "0910", "name": "9" },
      { "id": "1010", "name": "10" },
      { "id": "1110", "name": "11" },
      { "id": "1210", "name": "12" },
      { "id": "1310", "name": "13" },
      { "id": "1410", "name": "14" },
      { "id": "1510", "name": "15" },
      { "id": "1610", "name": "16" },
      { "id": "1710", "name": "17" },
      { "id": "1810", "name": "18" },
      { "id": "1910", "name": "19" },
      { "id": "2010", "name": "20" },
      { "id": "2110", "name": "21" },
      { "id": "2210", "name": "22" },
      { "id": "2310", "name": "23" },
      { "id": "2410", "name": "24" },
      { "id": "2510", "name": "25" },
      { "id": "2610", "name": "26" },
      { "id": "2710", "name": "27" },
      { "id": "2810", "name": "28" },
      { "id": "2910", "name": "29" },
      { "id": "3010", "name": "30" },
      { "id": "3110", "name": "31" }
    ]
  },
  {
    "id": "Nov",
    "name": "November",
    "days": [
      { "id": "0111", "name": "1" },
      { "id": "0211", "name": "2" },
      { "id": "0311", "name": "3" },
      { "id": "0411", "name": "4" },
      { "id": "0511", "name": "5" },
      { "id": "0611", "name": "6" },
      { "id": "0711", "name": "7" },
      { "id": "0811", "name": "8" },
      { "id": "0911", "name": "9" },
      { "id": "1011", "name": "10" },
      { "id": "1111", "name": "11" },
      { "id": "1211", "name": "12" },
      { "id": "1311", "name": "13" },
      { "id": "1411", "name": "14" },
      { "id": "1511", "name": "15" },
      { "id": "1611", "name": "16" },
      { "id": "1711", "name": "17" },
      { "id": "1811", "name": "18" },
      { "id": "1911", "name": "19" },
      { "id": "2011", "name": "20" },
      { "id": "2111", "name": "21" },
      { "id": "2211", "name": "22" },
      { "id": "2311", "name": "23" },
      { "id": "2411", "name": "24" },
      { "id": "2511", "name": "25" },
      { "id": "2611", "name": "26" },
      { "id": "2711", "name": "27" },
      { "id": "2811", "name": "28" },
      { "id": "2911", "name": "29" },
      { "id": "3011", "name": "30" }
    ]
  },
  {
    "id": "Dec",
    "name": "December",
    "days": [
      { "id": "0112", "name": "1" },
      { "id": "0212", "name": "2" },
      { "id": "0312", "name": "3" },
      { "id": "0412", "name": "4" },
      { "id": "0512", "name": "5" },
      { "id": "0612", "name": "6" },
      { "id": "0712", "name": "7" },
      { "id": "0812", "name": "8" },
      { "id": "0912", "name": "9" },
      { "id": "1012", "name": "10" },
      { "id": "1112", "name": "11" },
      { "id": "1212", "name": "12" },
      { "id": "1312", "name": "13" },
      { "id": "1412", "name": "14" },
      { "id": "1512", "name": "15" },
      { "id": "1612", "name": "16" },
      { "id": "1712", "name": "17" },
      { "id": "1812", "name": "18" },
      { "id": "1912", "name": "19" },
      { "id": "2012", "name": "20" },
      { "id": "2112", "name": "21" },
      { "id": "2212", "name": "22" },
      { "id": "2312", "name": "23" },
      { "id": "2412", "name": "24" },
      { "id": "2512", "name": "25" },
      { "id": "2612", "name": "26" },
      { "id": "2712", "name": "27" },
      { "id": "2812", "name": "28" },
      { "id": "2912", "name": "29" },
      { "id": "3012", "name": "30" },
      { "id": "3112", "name": "31" }
    ]
  }
]
