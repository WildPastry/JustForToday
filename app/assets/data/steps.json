[
  {
    "id": "1",
    "step": "We admitted we were powerless over alcohol â€” that our lives had become unmanageable."
  },
  {
    "id": "2",
    "step": "Came to believe that a Power greater than ourselves could restore us to sanity."
  },
  {
    "id": "3",
    "step": "Made a decision to turn our will and our lives over to the care of God as we understood Him."
  },
  {
    "id": "4",
    "step": "Made a searching and fearless moral inventory of ourselves."
  },
  {
    "id": "5",
    "step": "Admitted to God, to ourselves, and to another human being the exact nature of our wrongs."
  },
  {
    "id": "6",
    "step": "Were entirely ready to have God remove all these defects of character."
  },
  {
    "id": "7",
    "step": "Humbly asked Him to remove our shortcomings."
  },
  {
    "id": "8",
    "step": "Made a list of all persons we had harmed, and became willing to make amends to them all."
  },
  {
    "id": "9",
    "step": "Made direct amends to such people wherever possible, except when to do so would injure them or others."
  },
  {
    "id": "10",
    "step": "Continued to take personal inventory and when we were wrong promptly admitted it."
  },
  {
    "id": "11",
    "step": "Sought through prayer and meditation to improve our conscious contact with God as we understood Him, praying only for knowledge of His will for us and the power to carry that out."
  },
  {
    "id": "12",
    "step": "Having had a spiritual awakening as the result of these Steps, we tried to carry this message to alcoholics, and to practice these principles in all our affairs."
  }
]
