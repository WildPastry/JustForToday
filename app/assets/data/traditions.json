{
  "short": [
    {
      "id": "1",
      "tradition": "Our common welfare should come first; personal recovery depends upon A.A. unity."
    },
    {
      "id": "2",
      "tradition": "For our group purpose there is but one ultimate authority — a loving God as He may express Himself in our group conscience. Our leaders are but trusted servants; they do not govern."
    },
    {
      "id": "3",
      "tradition": "The only requirement for A.A. membership is a desire to stop drinking."
    },
    {
      "id": "4",
      "tradition": "Each group should be autonomous except in matters affecting other groups or A.A. as a whole."
    },
    {
      "id": "5",
      "tradition": "Each group has but one primary purpose — to carry its message to the alcoholic who still suffers."
    },
    {
      "id": "6",
      "tradition": "An A.A. group ought never endorse, finance, or lend the A.A. name to any related facility or outside enterprise, lest problems of money, property, and prestige divert us from our primary purpose."
    },
    {
      "id": "7",
      "tradition": "Every A.A. group ought to be fully self-supporting, declining outside contributions."
    },
    {
      "id": "8",
      "tradition": "Alcoholics Anonymous should remain forever non-professional, but our service centers may employ special workers."
    },
    {
      "id": "9",
      "tradition": "A.A., as such, ought never be organized; but we may create service boards or committees directly responsible to those they serve."
    },
    {
      "id": "10",
      "tradition": "Alcoholics Anonymous has no opinion on outside issues; hence the A.A. name ought never be drawn into public controversy."
    },
    {
      "id": "11",
      "tradition": "Our public relations policy is based on attraction rather than promotion; we need always maintain personal anonymity at the level of press, radio, and films."
    },
    {
      "id": "12",
      "tradition": "Anonymity is the spiritual foundation of all our traditions, ever reminding us to place principles before personalities."
    }
  ],
  "long": [
    {
      "id": "1",
      "tradition": "Each member of Alcoholics Anonymous is but a small part of a great whole. A.A. must continue to live or most of us will surely die. Hence our common welfare comes first. But individual welfare follows close afterward."
    },
    {
      "id": "2",
      "tradition": "For our group purpose there is but one ultimate authority–a loving God as He may express Himself in our group conscience."
    },
    {
      "id": "3",
      "tradition": "Our membership ought to include all who suffer from alcoholism. Hence we may refuse none who wish to recover. Nor ought A.A. membership ever depend upon money or conformity. Any two or three alcoholics gathered together for sobriety may call themselves an A.A. group, provided that, as a group, they have no other affiliation."
    },
    {
      "id": "4",
      "tradition": "With respect to its own affairs, each A.A. group should be responsible to no other authority than its own conscience. But when its plans concern the welfare of neighboring groups also, those groups ought to be consulted. And no group, regional committee, or individual should ever take any action that might greatly affect A.A. as a whole without conferring with the Trustees of the General Service Board. On such issues our common welfare is paramount."
    },
    {
      "id": "5",
      "tradition": "Each Alcoholics Anonymous group ought to be a spiritual entity having but one primary purpose–that of carrying its message to the alcoholic who still suffers."
    },
    {
      "id": "6",
      "tradition": "Problems of money, property, and authority may easily divert us from our primary spiritual aim. We think, therefore, that any considerable property of genuine use to A.A. should be separately incorporated and managed, thus dividing the material from the spiritual. An A.A. group, as such, should never go into business. Secondary aids to A.A., such as clubs or hospitals which require much property or administration, ought to be incorporated and so set apart that, if necessary, they can be freely discarded by the groups. Hence such facilities ought not to use the A.A. name. Their management should be the sole responsibility of those people who financially support them. For clubs, A.A. managers are usually preferred. But hospitals, as well as other places of recuperation, ought to be well outside A.A.- and medically supervised. While an A.A. group may cooperate with anyone, such cooperation ought never go so far as affiliation or endorsement, actual or implied. An A.A. group can bind itself to no one."
    },
    {
      "id": "7",
      "tradition": "The A.A. groups themselves ought to be fully supported by the voluntary contributions of their own members. We think that each group should soon achieve this ideal; that any public solicitation of funds using the name of Alcoholics Anonymous is highly dangerous, whether by groups, clubs, hospitals, or other 2 outside agencies; that acceptance of large gifts from any source, or of contributions carrying any obligation whatever, is unwise. Then too, we view with much concern those A.A. treasuries which continue, beyond prudent reserves, to accumulate funds for no stated A.A. purpose. Experience has often warned us that nothing can so surely destroy our spiritual heritage as futile disputes over property, money, and authority."
    },
    {
      "id": "8",
      "tradition": "Alcoholics Anonymous should remain forever non-professional. We define professionalism as the occupation of counseling alcoholics for fees or hire. But we may employ alcoholics where they are going to perform those services for which we may otherwise have to engage nonalcoholics. Such special services may be well recompensed. But our usual A.A. '12th Step' work is never to be paid for."
    },
    {
      "id": "9",
      "tradition": "Each A.A. group needs the least possible organization. Rotating leadership is the best. The small group may elect its secretary, the large group its rotating committee, and the groups of a large metropolitan area their central or intergroup committee, which often employs a full-time secretary. The trustees of the General Service Board are, in effect, our A.A. General Service Committee. They are the custodians of our A.A. Tradition and the receivers of voluntary A.A. contributions by which we maintain our A.A. General Service Office at New York. They are authorized by the groups to handle our over-all public relations and they guarantee the integrity of our principle newspaper, the A.A. Grapevine. All such representatives are to be guided in the spirit of service, for true leaders in A.A. are but trusted and experienced servants of the whole. They derive no real authority from their titles; they do not govern. Universal respect is the key to their usefulness."
    },
    {
      "id": "10",
      "tradition": "No A.A. group or member should ever, in such a way as to implicate A.A., express any opinion on outside controversial issues–particularly those of politics, alcohol reform, or sectarian religion. The Alcoholics Anonymous groups oppose no one. Concerning such matters they can express no views whatever."
    },
    {
      "id": "11",
      "tradition": "Our relations with the general public should be characterized by personal anonymity. We think A.A. ought to avoid sensational advertising. Our names and pictures as A.A. members ought not be broadcast, filmed, or publicly printed. Our public relations should be guided by the principle of attraction rather than promotion. There is never need to praise ourselves. We feel it better to let our friends recommend us."
    },
    {
      "id": "12",
      "tradition": "And finally, we of Alcoholics Anonymous believe that the principle of anonymity has an immense spiritual significance. It reminds us that we are to place principles before personalities; that we are actually to practice a genuine humility. This to the end that our great blessings may never spoil us; that we shall forever live in thankful contemplation of Him who presides over us all."
    }
  ]
}
