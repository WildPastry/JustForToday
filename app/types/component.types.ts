export interface ExternalLink {
  children: JSX.Element;
  href: string;
  style: {};
}

export interface EditScreenInfo {
  path: string;
}
