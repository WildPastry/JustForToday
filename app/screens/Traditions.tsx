import { ColorSchemeName, Pressable, StyleSheet } from 'react-native';
import { ETraditionTypes, ITraditions } from '../types/data.types';
import { ForwardedScrollView, Text, View } from '../components/styles/Themed';
import React, { useRef, useState } from 'react';
import { AppState } from '../redux/store';
import Colours from '../constants/Colours';
import { FontAwesome } from '@expo/vector-icons';
import { FontDisplay } from '../components/styles/StyledText';
import Tradition from '../components/layout/Tradition';
import { useAppSelector } from '../redux/hooks';
import useColorScheme from '../hooks/useColorScheme';
import { useFocusEffect } from '@react-navigation/native';

const Traditions: React.FC = (): JSX.Element => {
  // Screen settings
  const scrollViewRef: React.MutableRefObject<any> = useRef<any>(null);
  const colorScheme: NonNullable<ColorSchemeName> = useColorScheme();
  const [traditionType, setTraditionType] = useState(ETraditionTypes.short);

  // Data from store
  const traditions: ITraditions = useAppSelector(
    (state: AppState): ITraditions => {
      return state.data.traditions;
    }
  );

  useFocusEffect(
    React.useCallback(() => {
      // Scroll to top on focus
      scrollViewRef.current?.scrollTo({ y: 0, animated: false });
      return () => null;
    }, [scrollViewRef])
  );

  const isSelected = (activeTraditionType: ETraditionTypes): boolean => {
    return activeTraditionType === traditionType;
  };

  return (
    <ForwardedScrollView
      contentContainerStyle={styles.container}
      ref={scrollViewRef}>
      <View style={styles.logoContainer}>
        {/* Logo */}
        <FontAwesome
          style={styles.text}
          name='book'
          size={25}
          color={Colours[colorScheme].text}
        />
        {/* Title */}
        <FontDisplay style={styles.title}>Traditions</FontDisplay>
      </View>
      {/* Divider */}
      <View
        style={styles.divider}
        lightColor={Colours.light.seperator}
        darkColor={Colours.dark.seperator}
      />
      {/* Controls */}
      <View style={styles.controls}>
        <Pressable
          style={[
            styles.traditionBtn,
            isSelected(ETraditionTypes.short)
              ? { backgroundColor: Colours[colorScheme].currentBtn }
              : { backgroundColor: Colours[colorScheme].btn }
          ]}
          onPress={() => setTraditionType(ETraditionTypes.short)}>
          <Text
            style={[
              styles.text,
              isSelected(ETraditionTypes.short)
                ? { color: Colours[colorScheme].currentBtnText }
                : null
            ]}>
            SHORT
          </Text>
        </Pressable>
        <Pressable
          style={[
            styles.traditionBtn,
            isSelected(ETraditionTypes.long)
              ? { backgroundColor: Colours[colorScheme].currentBtn }
              : { backgroundColor: Colours[colorScheme].btn }
          ]}
          onPress={() => setTraditionType(ETraditionTypes.long)}>
          <Text
            style={[
              styles.text,
              isSelected(ETraditionTypes.long)
                ? { color: Colours[colorScheme].currentBtnText }
                : null
            ]}>
            LONG
          </Text>
        </Pressable>
      </View>
      {/* Traditions */}
      {traditions[traditionType].map((tradition, index) => (
        <Tradition
          key={index}
          id={tradition.id}
          tradition={tradition.tradition}
        />
      ))}
    </ForwardedScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 15
  },
  logoContainer: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'center',
    marginTop: 10
  },
  controls: {
    flexDirection: 'row',
    gap: 20,
    marginBottom: 20,
    marginTop: 10
  },
  title: {
    fontSize: 20,
    marginLeft: 10
  },
  text: {
    textAlign: 'center'
  },
  divider: {
    alignSelf: 'center',
    marginVertical: 20,
    height: 1,
    width: '70%'
  },
  traditionBtn: {
    flex: 1,
    borderRadius: 12,
    paddingVertical: 12
  }
});

export default Traditions;
