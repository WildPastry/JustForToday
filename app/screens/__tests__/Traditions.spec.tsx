import { ITraditions } from '../../types/data.types';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux';
import Traditions from '../Traditions';
import { configureStore } from '@reduxjs/toolkit';
import renderer from 'react-test-renderer';

describe('<Traditions />', () => {
  it('renders correctly', () => {
    // Mock data
    const mockTraditions: ITraditions = {
      short: [
        { id: '1', tradition: 'Short tradition 1' },
        { id: '2', tradition: 'Short tradition 2' }
      ],
      long: [
        { id: '3', tradition: 'Long tradition 1' },
        { id: '4', tradition: 'Long tradition 2' }
      ]
    };

    // Create mock store with initial state including mock traditions data
    const store = configureStore({
      reducer: () => ({ data: { traditions: mockTraditions } })
    });

    const tree = renderer
      .create(
        <Provider store={store}>
          <NavigationContainer>
            <Traditions />
          </NavigationContainer>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
