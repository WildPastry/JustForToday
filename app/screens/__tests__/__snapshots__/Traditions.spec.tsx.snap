// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Traditions /> renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "padding": 20,
    }
  }
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <View>
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 10,
          },
        ]
      }
    >
      <Text />
      <Text
        style={
          [
            {
              "color": "#000",
            },
            [
              {
                "fontSize": 20,
                "marginLeft": 10,
              },
              {
                "fontFamily": "font-display",
              },
            ],
          ]
        }
      >
        Traditions
      </Text>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#bdc3c7",
          },
          {
            "alignSelf": "center",
            "height": 1,
            "marginVertical": 20,
            "width": "70%",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "flexDirection": "row",
            "gap": 20,
            "marginBottom": 20,
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "borderRadius": 12,
              "flex": 1,
              "paddingVertical": 12,
            },
            {
              "backgroundColor": "#2b5b7b",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#000",
              },
              [
                {
                  "fontSize": 15,
                  "lineHeight": 21,
                  "textAlign": "center",
                },
                {
                  "color": "#fff",
                },
              ],
            ]
          }
        >
          Short
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "borderRadius": 12,
              "flex": 1,
              "paddingVertical": 12,
            },
            {
              "backgroundColor": "#e5edf9",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#000",
              },
              [
                {
                  "fontSize": 15,
                  "lineHeight": 21,
                  "textAlign": "center",
                },
                null,
              ],
            ]
          }
        >
          Long
        </Text>
      </View>
    </View>
    <Text
      style={
        [
          {
            "color": "#000",
          },
          {
            "flexDirection": "row",
            "marginBottom": 20,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000",
            },
            {
              "fontSize": 15,
              "fontWeight": "200",
              "lineHeight": 21,
              "marginRight": 10,
            },
          ]
        }
      >
        1
          
      </Text>
      <Text
        style={
          [
            {
              "color": "#000",
            },
            {
              "fontSize": 15,
              "lineHeight": 21,
              "textAlign": "left",
            },
          ]
        }
      >
        Short tradition 1
      </Text>
    </Text>
    <Text
      style={
        [
          {
            "color": "#000",
          },
          {
            "flexDirection": "row",
            "marginBottom": 20,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000",
            },
            {
              "fontSize": 15,
              "fontWeight": "200",
              "lineHeight": 21,
              "marginRight": 10,
            },
          ]
        }
      >
        2
          
      </Text>
      <Text
        style={
          [
            {
              "color": "#000",
            },
            {
              "fontSize": 15,
              "lineHeight": 21,
              "textAlign": "left",
            },
          ]
        }
      >
        Short tradition 2
      </Text>
    </Text>
  </View>
</RCTScrollView>
`;
